redirects/redirect_middleware.py

```diff
@@ -0,0 +1,38 @@
+# written based on https://github.com/django/django/blob/stable/3.2.x/django/contrib/redirects/middleware.py
+# since the part of the middleware that looks up the redirect from the 
+# database is not encapsulated, we cannot easily subclass this the
+# RedirectFallbackMiddleware. 
+
+from django.http import HttpResponseGone, HttpResponsePermanentRedirect
+from django.utils.deprecation import MiddlewareMixin
+from .create_redirect_dict import get_redirects
+
+REDIRECTS = get_redirects()
```

Comment:
Would it make more sense to load this into the global configs as part of the app initialization with [AppConfig.ready()](https://docs.djangoproject.com/en/5.0/ref/applications/#django.apps.AppConfig.ready)

Reply:
Hmmm, I understood this as read once when settings.py initializes middleware, but yes, putting it in AppConfig.ready() certainly makes that less ambiguous.